lib/versions.js

```diff
@@ -1,22 +1,54 @@
 import fs from "fs/promises";
 import * as core from "@actions/core";
+import * as exec from "@actions/exec";
 import { HttpClient } from "@actions/http-client";
 
-export async function determineVersion(repo, versionInput) {
-  versionInput = versionInput?.trim();
+export async function determineVersion(version, { repo, nightliesRepo }) {
+  version = version?.trim();
 
-  if (!versionInput) {
+  if (!version) {
     let toolVersion = await getVersionFromToolVersionsFile();
-    versionInput = toolVersion ?? "latest";
+    version = toolVersion ?? "latest";
   }
 
-  if (versionInput === "latest") {
-    versionInput = await fetchLatestTag(repo);
+  if (version === "latest") {
+    version = await fetchLatestTag(repo);
   }
 
-  return versionInput.startsWith("v")
-    ? versionInput.substring(1)
-    : versionInput;
+  if (version === "nightly") {
+    version = await fetchLatestTag(nightliesRepo);
+  }
+
+  if (version.startsWith("v")) {
+    version = version.substring(1);
+  }
+
+  if (version.startsWith("nightly-")) {
+    return {
+      repo: nightliesRepo,
+      version,
+    };
+  } else {
+    return {
+      repo,
+      version,
+    };
+  }
+}
```

Comment:
Nit: We could avoid branching by using ternary operator for `repo`

```suggestion
return {
  repo: version.startsWith("nightly-") ? nightliesRepo : repo,
  version,
};
```